core/src/main/scala/io/odin/formatter/Formatter.scala

```diff
@@ -127,4 +157,24 @@ object Formatter {
         writeStackTrace(b, elements.tail, indent)
     }
   }
+
+  private def abbreviate(enclosure: String): String = {
+    @tailrec
+    def loop(input: Array[String], builder: StringBuilder): StringBuilder = {
+      input.length match {
+        case 0 => builder
+        case 1 => builder.append(input.head)
+        case _ =>
+          val head = input.head
+          val b = if (head.isEmpty) builder.append(questionMark) else builder.append(head.head)
+          loop(input.tail, b.append(dot))
+      }
+    }
+
+    loop(enclosure.split('.'), new StringBuilder).toString()
```

Comment:
Let's reuse `dot` here as well

Reply:
`split` treats string argument as a regex. The suitable options are `"\\."`, `'.'`.
Should I change it to:
```
loop(enclosure.split(abbreviateSeparator), new StringBuilder).toString()
private val abbreviateSeparator = "\\."
```
?

Reply:
Ah right, it's a special symbol. Nvm then, it's not worth it

Reply:
251a443